### PR TITLE
Mekhanik evgenii/1196 8

### PR DESCRIPTION
**Pull request by EvgeniiMekhanik**

### CODE REVIEW

**Comment by EvgeniiMekhanik** [comment]
Please review this PR after 1953

File: fw/http_frame.c [hunk 1]
```diff
@@ -2445,76 +2456,47 @@ do {									\
...
 		TfwStream *stream = rb_entry(node, TfwStream, node);
-		if (stream->xmit.skb_head)
+		TfwStreamSchedEntry *parent = stream->sched.parent;
+
+		if (stream->xmit.skb_head) {
+			tfw_h2_stream_schred_remove(stream);
 			tfw_h2_stream_purge_send_queue(stream);
+			tfw_h2_sched_stream_enqueue(stream, parent);
+		}
 		__tfw_h2_purge_stream_send_queue(node->rb_left);
```
Comment (EvgeniiMekhanik) [inline fw/http_frame.c:2533]: Is recursion is available for us? Kernel stack is small, so can we use recursion?

File: fw/http_stream_sched.c [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+		21173, 21065, 20958, 20852, 20748, 20644, 20541, 20439,
+		20339, 20239, 20140, 20043, 19946, 19850, 19755, 19661,
+		19568, 19475, 19384, 19293, 19204, 19115, 19027, 18939,
+		18853, 18767, 18682, 18598, 18515, 18432, 18350, 18269,
+		18188, 18109, 18030, 17951, 17873, 17796, 17720, 17644,
+		17569, 17495, 17421, 17348, 17275, 17203, 17132, 17061,
+		16991, 16921, 16852, 16784, 16716, 16648, 16581, 16515,
+		16449, 16384, 16319, 16255, 16191, 16128
+	};
+	TfwStreamSchedEntryAnchor *anchor;
```
Comment (const-t) [inline fw/http_stream_sched.c:323]: `anchor` is unused

File: fw/http_stream_sched.c [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+				  1 - stream->link.offset));
+		}
+
+		stream = next;
+		parent = stream->sched.parent;
+	}
+}
+
+void
+tfw_h2_stream_schred_remove(TfwStream *stream)
```
Comment (const-t) [inline fw/http_stream_sched.c:86]:
```suggestion
tfw_h2_stream_sched_remove(TfwStream *stream)
```

**Pull request closed without merging**